src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/ScannedDocumentsHelper.java

```diff
@@ -44,6 +47,61 @@ private ScannedDocumentsHelper() {
             .collect(toList());
     }
 
+    public static void setExceptionRecordIdToScannedDocuments(
+        ExceptionRecord exceptionRecord,
+        CaseUpdateDetails caseDetails
+    ) {
+        @SuppressWarnings("unchecked")
+        var caseData = (Map<String, Object>) caseDetails.caseData;
+        List<ScannedDocument> scannedDocuments = getScannedDocuments(caseData);
+
+        if (exceptionRecord.scannedDocuments == null) {
+            // ensure scanned documents not null
+            caseData.put(SCANNED_DOCUMENTS, scannedDocuments);
+            return;
+        }
+
+        List<String> exceptionRecordDcns = exceptionRecord.scannedDocuments
+            .stream()
+            .map(scannedDocument -> scannedDocument.controlNumber)
+            .collect(toList());
+
+        if (exceptionRecordDcns.isEmpty()) {
+            // ensure scanned documents not null
+            caseData.put(SCANNED_DOCUMENTS, scannedDocuments);
+            return;
+        }
+
+        List<ScannedDocument> updatedScannedDocuments = scannedDocuments.stream()
+            .map(scannedDocument ->
+                exceptionRecordDcns.contains(scannedDocument.controlNumber)
+                    ? new ScannedDocument(
+                        scannedDocument.fileName,
+                        scannedDocument.controlNumber,
+                        scannedDocument.type,
+                        scannedDocument.subtype,
+                        scannedDocument.scannedDate,
+                        scannedDocument.url,
+                        scannedDocument.deliveryDate,
+                        exceptionRecord.id
+                      )
```

Comment:
why are we creating a new scanned doc ?

Reply:
Because there are not setters in ScannedDocument class

Reply:
won't it result into duplicate scanned documents one with exception record id and one without

Reply:
There are unit tests to ensure won't

Reply:
But why do we need to create new scanned document and not just update ExceptionRecord reference in the newly added scanned document. I think currently only scanned documents are getting adding so we just to set one extra field there

Reply:
Because all fields in _ScannedDocument_ class are final, it cannot be updated even if it is public.